pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -787,3 +789,147 @@ def longest_common_subsequence(seq1: OneDimensionalArray, seq2: OneDimensionalAr
                     check_mat[i][j] = check_mat[i][j-1]
 
     return OneDimensionalArray(seq1._dtype, check_mat[row][col][-1])
+
+'''Tim Sort Implementation'''
+
+def insertion_sort(array, left=0, right=None):
+    '''
+    
+    This insertion sort function is required for implementation of timsort
+    In Insertion sort, The array is virtually split into a sorted and an 
+    unsorted part. Values from the unsorted part are picked and placed at
+    the correct position in the sorted part.The first step in implementing
+    Timsort is modifying the implementation of insertion_sort() from before
+    : This modified implementation adds a couple of parameters, left and 
+    right, that indicate which portion of the array should be sorted. This
+    allows the Timsort algorithm to sort a portion of the array in place. 
+    Modifying the function instead of creating a new one means that it can 
+    be reused for both insertion sort and Timsort.
+    
+  
+    '''
```

Comment:
Add parameters, examples, references  in the documentation

Reply:
I've added parameters, examples, references already

Reply:
All the docstrings should be at the start of the function. Please take a look at how the above functions are implemented, after that make the commit similar to it.